File: code/language/src/main/java/org/fbme/ide/iec61499/repository/PlatformRepositoryProvider.java
package org.fbme.ide.iec61499.repository;

import jetbrains.mps.project.Project;

import java.util.HashMap;
import java.util.Map;

public class PlatformRepositoryProvider {

  private static final Map<Project, PlatformRepository> ourRepositories = new HashMap<>();

  public static PlatformRepository getInstance(Project project) {
    return ourRepositories.get(project);
  }

  public static void init(Project project) {
    ourRepositories.put(project, new PlatformRepository(project.getRepository()));
  }

  public static void dispose(Project project) {
    ourRepositories.remove(project).dispose();
  }
}

File: code/language/src/main/kotlin/org/fbme/ide/iec61499/repository/PlatformRepositoryProvider.kt
package org.fbme.ide.iec61499.repository

import jetbrains.mps.project.Project
import org.jetbrains.mps.openapi.module.SRepository

object PlatformRepositoryProvider {
    private val ourRepositories: MutableMap<SRepository, PlatformRepository> = HashMap()

    @JvmStatic
    fun getInstance(project: Project) = getInstance(project.repository)

    @JvmStatic
    fun getInstance(repository: SRepository): PlatformRepository {
        return ourRepositories[repository] ?: error("PlatformRepository not found")
    }

    @JvmStatic
    fun init(project: Project) {
        val repository = project.repository
        ourRepositories[repository] = PlatformRepository(repository)
    }

    @JvmStatic
    fun dispose(project: Project) {
        ourRepositories.remove(project.repository)!!.dispose()
    }
}
